### DISCUSSION TITLE
EEPROM - writing smaller sections of data

### DISCUSSION THREAD

**Discussion by coledeck** [earlephilhower/arduino-pico#1775] [General]
Hi, 
I'm using the simulated flash EEPROM to "cache" some data. It's pretty much write once, read many, for 8 byte chunks. 

Looking at the EEPROM code, it looks like if you change the EEPROM at all, when running commit() it wipes and flashes the whole thing.
In order to help out the flash, is it possible to only write sections of arbitrary # of bytes / writing only different bytes? In my case, that would mean writing to each segment of flash only once, instead of every time I add anything to the EEPROM.

**Comment by earlephilhower** [comment]
Your assessment is correct.  There is no partial page programming in EEPROM.  If you care about flash lifetime, look at LittleFS which will wear level over the whole filesystem.

**Reply by coledeck** [reply]
Makes sense. In theory though there's nothing stopping it from being added because it's not a real EEPROM right?

Also - unless I'm missing something - it looks like it doesn't clear the dity bit after commit()?

**Reply by earlephilhower** [reply]
I think you're right.  Want to fix it with a PR?

**Reply by coledeck** [reply]
Will do!

**Comment by coledeck** [comment]
#1776 for dirty bit bugfix
